config/initializers/saml_idp.rb

```diff
@@ -45,3 +45,5 @@
     ServiceProvider.new(issuer_or_entity_id).metadata
   end
 end
+
+VALID_AUTHNCONTEXTS = JSON.parse(Figaro.env.valid_authncontexts).freeze
```

Comment:
+1 to @pkarman comment. And can we also make `AUTHNCONTEXTS` into two words for readability? (`Saml::Idp::Constants::VALID_AUTHN_CONTEXTS`)